### PR TITLE
feat(recording): StartRecordingDialogContent web and native

### PR DESCRIPTION
**Pull request by Calinteodor**
1. Created AbstractStartRecordingDialogContent.tsx.
2. Created ./features/recording/web/StartRecordingDialogContent.tsx.
3. Created ./features/recording/native/StartRecordingDialogContent.tsx.
4. Deleted old StartRecordingDialogContent.

### CODE REVIEW

File: react/features/recording/components/Recording/AbstractStartRecordingDialogContent.tsx [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+// @ts-ignore
+import { authorizeDropbox, updateDropboxToken } from '../../../dropbox';
+// @ts-ignore
+import { isVpaasMeeting } from '../../../jaas/functions';
+// @ts-ignore
+import { RECORDING_TYPES } from '../../constants';
+// @ts-ignore
+import { supportsLocalRecording } from '../../functions';
+
+export type Props = {
```
Comment (robertpin) [inline react/features/recording/components/Recording/AbstractStartRecordingDialogContent.tsx:29]: This should be an interface that extends WithTranslation

File: react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx [hunk 2]
```diff
@@ -0,0 +1,339 @@
+/* eslint-disable lines-around-comment  */
+import React from 'react';
+import { Image, View } from 'react-native';
+import { Switch, Text } from 'react-native-paper';
+
+// @ts-ignore
+import { translate } from '../../../../base/i18n';
```
Comment (robertpin) [inline react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx:7]: Add /functions to the import path and remove @ts-ignore

File: react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+// @ts-ignore
+import { LoadingIndicator } from '../../../../base/react';
+// @ts-ignore
+import { connect } from '../../../../base/redux';
```
Comment (robertpin) [inline react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx:11]: Add /functions to the import path and remove @ts-ignore

File: react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+import Button from '../../../../base/ui/components/native/Button';
+import { BUTTON_TYPES } from '../../../../base/ui/constants';
+// @ts-ignore
+import { RECORDING_TYPES } from '../../../constants';
+// @ts-ignore
+import { getRecordingDurationEstimation } from '../../../functions';
+import AbstractStartRecordingDialogContent, {
+    type Props,
```
Comment (robertpin) [inline react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx:19]: In TS you can just import Props, no need to specify "type"

File: react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+    TRACK_COLOR
+    // @ts-ignore
+} from '../styles.native';
+
+
+/**
+ * The start recording dialog content for the mobile application.
+ */
+class StartRecordingDialogContent extends
+// @ts-ignore
```
Comment (robertpin) [inline react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx:36]: Why is this needed?

File: react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+    AbstractStartRecordingDialogContent<Props> {
+    /**
+     * Renders the component.
+     *
+     * @protected
+     * @returns {React$Component}
+     */
+    render() {
+        // @ts-ignore
```
Comment (robertpin) [inline react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx:45]: AFAIK this is needed only of the Abstract class is not TS. But the Abstract class used here is TS so why is this needed?

File: react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+            <View style = { styles.container }>
+                { this._renderNoIntegrationsContent() }
+                { this._renderFileSharingContent() }
+                { this._renderUploadToTheCloudInfo() }
+                { this._renderIntegrationsContent() }
+            </View>
+        );
+    }
+
+    _shouldRenderNoIntegrationsContent: () => boolean;
```
Comment (robertpin) [inline react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx:58]: Why are these needed? They should be removed in TS files

File: react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx [hunk 8]
```diff
@@ -0,0 +1,429 @@
+/* eslint-disable lines-around-comment  */
+import React from 'react';
+
+// @ts-ignore
+import { translate } from '../../../../base/i18n';
```
Comment (robertpin) [inline react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx:5]: Add `/functions` to import path and remove @ts-ignore

File: react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx [hunk 9]
```diff
@@ -0,0 +1,429 @@
...
+import {
+    Container,
+    Image,
+    LoadingIndicator,
+    Switch,
+    Text
+    // @ts-ignore
+} from '../../../../base/react';
+// @ts-ignore
+import { connect } from '../../../../base/redux';
```
Comment (robertpin) [inline react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx:16]: Add `/functions` to import path and remove @ts-ignore

File: react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx [hunk 10]
```diff
@@ -0,0 +1,429 @@
...
+import Button from '../../../../base/ui/components/web/Button';
+import { BUTTON_TYPES } from '../../../../base/ui/constants';
+// @ts-ignore
+import { RECORDING_TYPES } from '../../../constants';
+// @ts-ignore
+import { getRecordingDurationEstimation } from '../../../functions';
+import AbstractStartRecordingDialogContent, {
+    type Props,
```
Comment (robertpin) [inline react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx:24]: In TS you can just import Props, no need to specify "type"

File: react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx [hunk 11]
```diff
@@ -0,0 +1,429 @@
...
+    TRACK_COLOR
+    // @ts-ignore
+} from '../styles.web';
+
+
+/**
+ * The start recording dialog content for the mobile application.
+ */
+class StartRecordingDialogContent extends
+// @ts-ignore
```
Comment (robertpin) [inline react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx:42]: Why is this needed?

File: react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx [hunk 12]
```diff
@@ -0,0 +1,429 @@
...
+                { this._renderNoIntegrationsContent() }
+                { this._renderFileSharingContent() }
+                { this._renderUploadToTheCloudInfo() }
+                { this._renderIntegrationsContent() }
+                { this._renderLocalRecordingContent() }
+            </Container>
+        );
+    }
+
+    _shouldRenderNoIntegrationsContent: () => boolean;
```
Comment (robertpin) [inline react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx:62]: Ditto

File: react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx [hunk 13]
```diff
@@ -0,0 +1,429 @@
...
+        }
+
+        const {
+            _localRecordingEnabled,
+            integrationsEnabled,
+            isValidating,
+            isVpaas,
+            selectedRecordingService,
+            t
+            // @ts-ignore
```
Comment (robertpin) [inline react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx:84]: Ditto

**Review by robertpin — CHANGES_REQUESTED** [review]

File: react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx (hunk 5 above)
Comment (Calinteodor) [reply react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx:36]: Type 'AbstractStartRecordingDialogContent' is not generic.

File: react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx (hunk 6 above)
Comment (Calinteodor) [reply react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx:45]: Property 'props' does not exist on type

File: react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx (hunk 7 above)
Comment (Calinteodor) [reply react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx:58]: TS2551: Property '_shouldRenderNoIntegrationsContent' does not exist on type 'StartRecordingDialogContent'. 
Same for the others

File: react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx (hunk 13 above)
Comment (Calinteodor) [reply react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx:84]: Property 'props' does not exist on type 'StartRecordingDialogContent'

File: react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx (hunk 5 above)
Comment (robertpin) [reply react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx:36]: That means that you tried to pass a type to it when it doesn't support it. Just remove `<Props>` from `AbstractAbstractStartRecordingDialogContent<Props>` or make the component generic by modifying AbstractStartRecordingDialogContent to receive a type for Props like `class AbstractStartRecordingDialogContent<P extends Props> extends Component<P>`

File: react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx (hunk 6 above)
Comment (robertpin) [reply react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx:45]: If you solve the comment above it should solve this one as well

File: react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx (hunk 5 above)
Comment (Calinteodor) [reply react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx:36]: Thanks for the hint!

File: react/features/recording/components/Recording/AbstractStartRecordingDialogContent.tsx [hunk 14]
```diff
@@ -0,0 +1,360 @@
...
+            this._onLocalRecordingSwitchChange();
+        }
+    }
+
+    /**
+     * Implements {@code Component#componentDidUpdate}.
+     *
+     * @inheritdoc
+     */
+    componentDidUpdate(prevProps: Props) {
```
Comment (robertpin) [inline react/features/recording/components/Recording/AbstractStartRecordingDialogContent.tsx:189]: The type should be `P` not `Props`

File: react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx (hunk 7 above)
Comment (robertpin) [reply react/features/recording/components/Recording/native/StartRecordingDialogContent.tsx:53]: You should be able to remove them now that the Abstract component issue is solved

**Review by robertpin — CHANGES_REQUESTED** [review]

File: react/features/recording/components/Recording/AbstractStartRecordingDialogContent.tsx [hunk 15]
```diff
@@ -0,0 +1,360 @@
...
+/**
+ * Maps part of the redux state to the props of this component.
+ *
+ * @param {Object} state - The Redux state.
+ * @param {Object} ownProps - The own props of the component.
+ * @returns {Props}
+ */
+export function mapStateToProps(state: any, ownProps: Props) {
+    const localRecordingAvailable
+        = ownProps._localRecordingEnabled && supportsLocalRecording();
```
Comment (robertpin) [inline react/features/recording/components/Recording/AbstractStartRecordingDialogContent.tsx:346]: `_localRecordingEnabled` is not part of ownProps. It is declared below

File: react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx [hunk 16]
```diff
@@ -0,0 +1,401 @@
...
+            _localRecordingEnabled,
+            integrationsEnabled,
+            isValidating,
+            isVpaas,
+            selectedRecordingService,
+            t
+        } = this.props;
+
+        const switchContent
+            = integrationsEnabled || _localRecordingEnabled
```
Comment (robertpin) [inline react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx:77]: Please use `_localRecordingAvailable` instead of `_localRecordingEnabled`

File: react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx [hunk 17]
```diff
@@ -0,0 +1,401 @@
...
+                        className = 'recording-switch'
+                        disabled = { isValidating }
+                        onValueChange = { this._onRecordingServiceSwitchChange }
+                        trackColor = {{ false: TRACK_COLOR }}
+                        value = { selectedRecordingService === RECORDING_TYPES.JITSI_REC_SERVICE } />
+                ) : null;
+
+        const label = isVpaas ? t('recording.serviceDescriptionCloud') : t('recording.serviceDescription');
+        const jitsiContentRecordingIconContainer
+            = integrationsEnabled || _localRecordingEnabled
```
Comment (robertpin) [inline react/features/recording/components/Recording/web/StartRecordingDialogContent.tsx:89]: Same as above

**Review by robertpin — CHANGES_REQUESTED** [review]

**Review by robertpin — APPROVED** [review]
